lib/route/src/Obelisk/Route.hs

```diff
@@ -239,6 +240,17 @@ tryDecode (Encoder (Identity impl)) x = _encoderImpl_decode impl x
 encode :: Encoder Identity parse decoded encoded -> decoded -> encoded
 encode (Encoder (Identity impl)) x = _encoderImpl_encode impl x
 
+-- | Similar to 'encode' above, this is a convenience function to get directly
+-- the textual encoding of a route (which can be used, for example, in a href),
+-- given that one has an 'Encoder' to a 'PageName'.
+encodeToText :: forall a. Encoder Identity Identity a PageName -> a -> Text
+encodeToText e a =
+  let
+    e' :: Encoder Identity (Either Text) a PathQuery
+    e' = pageNameEncoder <<< hoistParse (pure . runIdentity) e
```

Comment:
Is the hoistParse actually necessary here? I'd expect that you can just use encode on pageNameEncoder <<< e without doing that bit.

Reply:
`e` has a type signature: ```Encoder Identity Identity a PageName```, while 
```
pageNameEncoder :: (Applicative check, MonadError Text parse) => Encoder check parse PageName PathQuery
```
So that `hoistParse (pure . runIdentity)` is required to change the `parse` monad to `Identity` to something that has `MonadError Text`. There are of course alternative approaches, which one would you prefer?

Reply:
ahhh, okay, nevermind then, looks good :)